src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,120 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_STORAGE_CAPACITY = 10;
+    private static final double STORAGE_GROW_MULTIPLIER = 1.5;
+    private static final String INDEX_OUT_OF_BOUNDS_MESSAGE = "Index is out of bounds";
+    private static final String NO_SUCH_ELEMENT_MESSAGE = "No such element found";
+    private int size;
+    private T[] storage;
+
+    public ArrayList() {
+        storage = (T[]) new Object[INITIAL_STORAGE_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        shouldStorageGrow(1);
+        storage[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        isIndexOutOfBoundsToAdd(index);
```

Comment:
better method name. prefix `is` in method names stands for returning boolean value from it. `check`- for validating and throwing exception if failed, which is exactly our case
```suggestion
        checkIndexForAdd(index);
```